lib/CMakeLists.txt

```diff
@@ -41,6 +41,28 @@ add_library(libmosquitto SHARED
 	util_mosq.c util_mosq.h
 	will_mosq.c will_mosq.h)
 
+#target for building static version of library
```

Comment:
I think it would be best here to define say

set(SRC logging_mosq.c logging_mosq.h ...)

add_library(libmosquitto SHARED ${SRC})
add_library(libmosquitto_static STATIC ${SRC})